packages/frontend/src/redux/slices/signMessage/index.js

```diff
@@ -0,0 +1,81 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { createSelector } from 'reselect';
+
+import { wallet } from '../../../utils/wallet';
+import { showCustomAlert } from '../../actions/status';
+import { selectAccountId } from '../account';
+
+const SLICE_NAME = 'signMessage';
+
+export const selectSignMessageSlice = (state) => state[SLICE_NAME];
+
+export const SIGN_MESSAGE_STATUS = {
+    IN_PROGRESS: 'sign-message-in-progress',
+    NEEDS_CONFIRMATION: 'sign-message-needs-confirmation',
+    SUCCESS: 'sign-message-success',
+};
+
+export const selectSignMessageStatus = createSelector(
+    [selectSignMessageSlice],
+    (signMessage) => signMessage.status
+);
+
+export const selectSignedRequest = createSelector(
+    [selectSignMessageSlice],
+    (signMessage) => signMessage.signedRequest || {}
+);
+
+export function addQueryParams(baseUrl, queryParams) {
```

Comment:
Can we please re-use/re-org this to so that we don't have do duplicate this code from the TX sign slice?

https://github.com/near/near-wallet/blob/master/packages/frontend/src/redux/slices/sign/index.js#L82